Classes/app.rb

```diff
@@ -29,43 +47,33 @@ def create_person
     print 'To create a student, press 1, to create a teacher, press 2 : '
     option = gets.chomp
 
+    print 'Enter age: '
+    age = gets.chomp.to_i
+    print 'Enter name: '
+    name = gets.chomp
+
     case option
     when '1'
-      create_student
+      create_student(age, name)
     when '2'
-      create_teacher
+      create_teacher(age, name)
     else
       puts 'Invalid input. Try again'
     end
   end
 
-  def create_student
-    puts 'Create a new student'
-    print 'Enter student age: '
-    age = gets.chomp.to_i
-    print 'Enter name: '
-    name = gets.chomp
+  def create_student(age, name)
     print 'Has parent permission? [Y/N]: '
     parent_permission = gets.chomp.downcase
-    case parent_permission
-    when 'n'
-      Student.new(age, name, parent_permission: false)
-      puts 'Student doesnt have parent permission, cant rent books'
-    when 'y'
-      student = Student.new(age, name, parent_permission: false)
-      @people << student
-      puts 'Student created successfully'
-    end
+    permission = parent_permission == 'y'
+    student = Student.new(age, name, parent_permission: permission)
+    @people << student
+    puts 'Student created successfully'
```

Comment:
- [ ] in this method you are printing to the console, getting user input and saving the data, this is not in line with the single responsibility rule please have a look at [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) for more details.